### PR TITLE
Eval add search box

### PR DESCRIPTION
**Pull request by Quintec**
Added search bar on top of student evalutions that filters the responses as you type

### CODE REVIEW

File: frontend/src/components/CourseModal/EvaluationResponses.tsx [hunk 1]
```diff
@@ -164,6 +164,26 @@ const EvaluationResponses: React.FC<{
 
   return (
     <div>
+      <StyledInput
+        id="filter-input"
+        type="text"
+        placeholder="Search evaluations..."
+        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (kevinhu) [inline frontend/src/components/CourseModal/EvaluationResponses.tsx:171]: Hmm we probably shouldn't modify the elements after they have been rendered here by accessing the DOM – can you refactor this so we store a state variable for the input text and filter the arrays before rendering them below?

Comment (Quintec) [reply frontend/src/components/CourseModal/EvaluationResponses.tsx:171]: How would you be able to filter the arrays before rendering them? Wouldn't that require refreshing and rewriting the DOM elements every time the filter changes? That seems far more inefficient than this - I'm not technically modifying the elements at all, just hiding/showing them

Comment (kevinhu) [reply frontend/src/components/CourseModal/EvaluationResponses.tsx:171]: Yes – you'd call `.filter` on the arrays. It should be efficient enough, it's what we do for the main search (see https://github.com/coursetable/coursetable/blob/master/frontend/src/contexts/searchContext.tsx)

Comment (kevinhu) [reply frontend/src/components/CourseModal/EvaluationResponses.tsx:171]: i.e. like this: https://gigi.nullneuron.net/gigilabs/filter-list-as-you-type-with-react/

**Comment by Quintec** [comment]
I also refactored the code a little to not have so much duplicate code in that section

**Comment by kevinhu** [comment]
Looks good! Could you also implement highlighting for the matched term? (Also upgraded part of our backend while testing – don't worry about it)

**Comment by kevinhu** [comment]
Also, if it's not too much trouble, a bit of padding on the bottom would be great!

Screen Shot 2021-11-04 at 23 28 35

**Comment by Quintec** [comment]
@kevinhu uh i can no longer start the backend
```
api_1             | $ npx prisma db pull && npx prisma generate
api_1             | Prisma schema loaded from prisma/schema.prisma
api_1             | Datasource "db": MySQL database "yaleplus" at "mysql:3306"
api_1             |
api_1             | Introspecting based on datasource defined in prisma/schema.prisma …
mysql_1           | 2021-11-05  4:18:41 5 [Warning] Aborted connection 5 to db: 'yaleplus' user: 'root' host: '172.18.0.6' (Got an error reading communication packets)
mysql_1           | 2021-11-05  4:18:41 4 [Warning] Aborted connection 4 to db: 'yaleplus' user: 'root' host: '172.18.0.6' (Got an error reading communication packets)
api_1             |
api_1             | ✔ Introspected 4 models and wrote them into prisma/schema.prisma in 186ms
api_1             |
api_1             | Run yarn prisma generate to generate Prisma Client.
api_1             |
api_1             | Prisma schema loaded from prisma/schema.prisma
api_1             | Error:
api_1             | Error: EACCES: permission denied, unlink '/app/node_modules/.prisma/client/index.d.ts'
api_1             |
api_1             |
api_1             | error Command failed with exit code 1.
api_1             | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by kevinhu** [comment]
Hmm try removing your docker containers and rebuilding?

**Comment by Quintec** [comment]
did not help

**Comment by kevinhu** [comment]
Can you delete the repo and rebuild from a fresh clone?

**Pull request merged**